app/src/main/java/com/github/polypoly/app/map/MapActivity.kt

```diff
@@ -360,42 +415,74 @@ class MapActivity : ComponentActivity() {
         return BitmapDrawable(context.resources, scaledBitmap)
     }
 
+    /**
+     * Initializes the location overlay and sets the location listener.
+     */
     private fun initLocationOverlay(mapView: MapView): MyLocationNewOverlay {
         val locationProvider = GpsMyLocationProvider(mapView.context)
         var lastLocation = Location("")
 
         val locationOverlay = object : MyLocationNewOverlay(locationProvider, mapView) {
             override fun onLocationChanged(location: Location?, provider: IMyLocationProvider?) {
                 super.onLocationChanged(location, provider)
-                updateAllDistances(mapView, GeoPoint(location))
+                mapViewModel.setCloseLocation(
+                    updateAllDistancesAndClosest(mapView, GeoPoint(location))
+                )
                 mapViewModel.addDistanceWalked(lastLocation.distanceTo(location!!))
                 lastLocation = locationProvider.lastKnownLocation
+                print("location: $location")
             }
         }
 
         locationOverlay.enableMyLocation()
         locationOverlay.enableFollowLocation()
         locationOverlay.runOnFirstFix {
+            print("First fix: ${locationOverlay.myLocation}")
             runOnUiThread {
-                updateAllDistances(mapView, locationOverlay.myLocation)
+                updateAllDistancesAndClosest(mapView, locationOverlay.myLocation)
                 mapView.controller.animateTo(locationOverlay.myLocation)
                 mapViewModel.resetDistanceWalked()
             }
         }
         return locationOverlay
     }
 
+    /**
+     * Returns all markers on the map.
+     */
     private fun markersOf(mapView: MapView): List<Marker> {
         return mapView.overlays.filterIsInstance<Marker>()
     }
 
-    private fun updateAllDistances(mapView: MapView, myLocation: GeoPoint) {
+    /**
+     * Updates the distance of all markers and returns the closest one.
+     *
+     * @return the closest location or null if there are no locations close enough to the player
+     */
+    private fun updateAllDistancesAndClosest(
```

Comment:
Maybe this could be renamed into `updateAllDistancesAndFindClosest` to make it clearer.
However, if there is a need to add an _and_ in the function name, that usually means that function does too many things. Here, is there a sufficient performance gain to do the closest lookup while doing the update, compared to doing this in two separate functions?

Reply:
I guess that 2n is still n so technically no, but in my brain that is a lot for a phone. If you really insist then I can split it into 2 different functions. No real opinion as both cases (long name and bad complexity) feel just as bad to me.

Reply:
Name changed as yours is definitely better

Reply:
Well the issue is that line 433 you have `updateAllDistancesAndFindClosest(mapView, locationOverlay.myLocation)` which doesn't use the closest location found, which to me reveals that you have two different behaviors here that you could separate into `updateDistances` and `findClosestLocation`.
On the performance side it just takes profit of a for-loop only ran once instead of twice so it should be benign considering we don't have many locations.

Reply:
The mistake was that runOnFirstFix was not using the output, accidental omission on my end.

Reply:
Ok this makes more sense now 👍